doc/release-notes-changelog: details on qspi xip extension (#17364)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

* Added documentation for :ref:`qspi_xip_split_image` functionality.
* Added a section in the sysbuild-related migration guide about the migration of :ref:`child_parent_to_sysbuild_migration_qspi_xip` from child/parent image to sysbuild.

> * Extended :ref:`qspi_xip_split_image` support to nRF52840 SoC and nRF91 series.

Reviewer: -nrf91

> * Extended :ref:`qspi_xip_split_image` to support MCUBoot's direct-xip mode.

Reviewer: *directXIP

* Updated the procedure for signing the application image built for booting by MCUboot in direct-XIP mode with revert support.
  Now, the Intel-Hex file of the application image automatically receives a confirmation flag.

> * Extended :ref:`qspi_xip_split_image` support to nRF52840 SoC and nRF91 series.
> * Extended :ref:`qspi_xip_split_image` to support MCUBoot's direct-XIP mode.

Reviewer 2: I think this should go as a sub-bullet under "added":

```suggestion
  * :ref:`QSPI XIP split image <qspi_xip_split_image>` support for nRF52840 SoC, nRF91 series, and MCUBoot's direct-xip mode.
```
"direct-xip" for consistency with other documentation
